### ISSUE TITLE
Provide Access to Force Transform Data Mid Simulation

### ISSUE THREAD

**Issue by PBI-David** [vega/vega#3528]
I'm trying to recreate the following **animated** chart in Vega. 

https://old.reddit.com/r/dataisbeautiful/comments/vp04dv/oc_the_worst_first_half_for_the_stock_market_in/

![image](https://user-images.githubusercontent.com/86846675/177214929-4d6b68a1-7764-4b5c-bf2e-b4e0f1256b07.png)

I know Vega doesn't natively support animations but it does have a timer signal which can be repurposed as per the clock examples. I have created the following where you can mouse click to start a timer, click again to stop the timer and double click to reset. The timer signal drives a filter transform which cycles through 7 datapoints one after another giving the illusion of animation.

[Editor](https://vega.github.io/editor/#/url/vega/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykSArJQBWENgDsQAGhAATONABONHJnaT0AQQAETJBBpRdSRdtgnM2uAA8kCHABs4U7QHc6MbcSQOGcbQ5ITHAOENoQMGyu4toMOBLaCGwMEP6RpIqU0iDusvRoACwADEUy8DRkWGgATCUyOEiysjTiZDUyskiYKKgA2qDidnBoIN1MTtnevgpo-SBQXeRsigCeI5rZBgBew6gAHDIt8tZoAIwyU35opSA4inDE1wC+UqALmEurIwBCmzQ7aAEAkO4mOZwuPiuqBudwez1e80WZGWa3QAGE-gDUNUCiCwahziBLrsYfdHtCXm8kSj1pjdgcQEcbO0iZDdoTYeSipTER9kV90L8ZNtdkC8czsRDpuDbmT4VS+TT0XSarjGaCJdUpVCOXKKQj3p9USANsL-vTxSdUABmbW7LWyuEEnmG-nGoUgEWA4Hq-G21nSyWO8mnF3UgUgDFmrE4y1of3Elmcs5hxUR02e81oBlMq1qxM2+p66qpo0-FWoMW+iX5tnxotOksG8PGqOZmNq3OFO314M1Utu2nRi3Vq0+gsO5M2gdKkAer2Vn1dys9oNT60ziNthex0eA1eTvUb5tp40Zhc5jVWgBsq91ToKm-dFary9vAZ1DfJTYVZeVw9VONUHfAt72-J8h3bEdlwAdlXABaMCUxPP85xfJcrzQOCP12RCv2Q39B3-KDAL3VBsILPC+2dABdGRMEUJBxAgAAzZYEFmUBMBWHBdk9OAAEc-HEKBhmFbpFEwGUIEwNgcDOOoQH0EZOm6EAeW43iRhYmgHA+RRshsO4VK6BgEFYK9tAAXissxknEfT1JonlBgQPixicAB9aAfDEnCIE43lUPPLN9iAwkC1JJ1uRQoi0IAxdwtXKKuQg4id07TCCWS-D9UI2cQqxS98QdUDcpi-KI3nULXyy0q62y6iKqCuLt1C3dl3qwMkLylqCorYqJQTBrDydUNYtnaqsVqv0D1y8bKtbCsOqy4bA1GkM0pNAagNrQN-SnH8+qq9DdtXA7iy2tqOzOnDe0OrbCugrLxxG3Lj0W8sEpmiVXvW96ruWzL8T+qENvjR6drIkCGp6x8JpO76MPxGHutyo7XVna77WBiVUc-aiMZbSCLyAiiGqoqcFuO58kbJhCeupzGtyB+m7tQSm9VDZy6MzMhBlCQLXL4qAThkOJVL43J6G0AhtGqdSEWFkZFAYSRVxYnxUhkCRAoeOAHIC9AoAcQwAGtsglxYRgAQlVyQnmcpWhhGKB7Mcu8dY0OZ9cNtAuJ49yaDcgz6JgRQ2GwCZKxKF4QCtj4VbVgAyZO3bV-SAB4YIAfnThy4FMABqbRTlQfPHJ5X3MCNgOtPQWRxhN83FfjnBJZGIonMd4UFicI25mV9AVh8iZ6MDkZTfEOATGyWQ2GQFp-bkLoelGIIvNHvydJCWQRkTYVlik1AGL8OPGNaPjykqKSXJd422AcZZsk0vjlmaAW54XpAl9QUBVLXh5OA3k+7bxoLvV2xNz5MTILsUAXBEAixbLUdSd83IjC9B0b+v9-6rxGEA7IO8HB73QF6aBl9ZilAEEUHmMgkDWBmH0UAyxwEOUnnAFiUle6+RGCPUBTkZDIEUGbAedd3I2C4SAFiEcOJ-xXmpdABC44GzdvIZeBtHJyKtKAUCscZCoh0bDWOccE5wNGBI5eRCSF7ieLYtBfFxBsHkAFce9dPQrAQEwR+hCZHLwAfgjewxlEiScWYjRhdLG6QcMvLertH7PxkFYyBp5UHi3btbORC54EVE-ugeIrgAAU1RtAACptCqTMpQEULhqgAEpNj8NIVmOOMkI5mzMQWHIMA6BBPEm0uAAB1GgeQYDL1AnHLYy5DGBhiiAFYbEoApBiY0uZsTEngOIfvOscdrALKWZknJPhXYnB2TEw50TjYnP0csnhw81lSI2dY5Mdj6IX1YuxQKr9tLLFEtkHwOAYA9CKJQB0bFFCiVEVIn5ItH6mzUYcfSXR1BGwdK0g2ZB8jQkoEUH0jFmj7NAEZAypCBKSQKRU8yIpamy3lqk0AYLfnoDFiAK0LK9kuM9AxdFmLgVHQZXxNY1zh7ss2Fy1omLuaOw0hPRRFjEm+Lkf49AjjnGtxUaE9RBdiWgB8DksZDV5jhIMnHPQqQp4dINQgYZsgJhxzYg5AAyqFaZOoBB2okJgAZcAKhVDkZ0rxmy7VRP1YGLpPTW4fGsMfeljyTKYEqczNYti0kdy0ZEiBTLW4GIeRmuZqDbHOSAA)

The chart is using the force transform to encode each datapoint and as the timer ticks, new data is filtered and the force transform moves each datum to the correct location. This all seems to be working but currently looks very janky for a couple of reasons: 

1. The start speed for the force transform is too rapid giving it a staccato appearance as it cycles through each of the datapoints. I have tried changing the alpha and velocity decay but neither give the control required. 
2. The start position for the beginning of each simulation of the force transform is currently fixed to the previous value of the node hardcoded in the sample table (centre of the screen for x and previous value for y). The node table generated by the force transform isn't accessible with the actual node's x and y until the simulation completes but obviously the filter moves on before this can happen. 

In summary, is it possible to expose the force transform data mid simulation? I'm open to other approaches if you have any ideas or is this just beyond current capabilities of Vega without some serious development effort?

**Comment by PBI-David** [comment]
Closing this as I think I need to use triggers but haven't yet worked it out.
